packages/ses/src/module-proxy.js

```diff
@@ -0,0 +1,82 @@
+// For brevity, in this file, the term "module" without qualification means
+// "module exports namespace proxy", the type that the
+// `compartment.module(fullSpecifier)` function returns.
+//
+// Compartments need a mechanism to link a module from one compartment
+// to another.
+// The procedure is to use `compartment.module(specifier)` to obtain the module
+// from one compartment before loading, linking, or executing the specified
+// module, and threading it into the module map of another compartment.
+// For this to be possible, it is necessary to model the module exports
+// namespace as a proxy that will treat all access to those exported properties
+// as reference errors until the properties of the actual module are known.
+// This provides the mechanism for modeling the public module namespace and
+// eventually connecting it to to the internal module exports namespace object.
+
+import { immutableObject } from './commons.js';
+import { makeAlias } from './module-load.js';
+
+const { freeze } = Object;
+// q, as in quote, for error messages.
+const q = JSON.stringify;
+
+// `createDeferredModule` creates a module and activator.
+// A `Compartment` can create a module for any
+// module specifier, regardless of whether it is loadable or executable, and
+// use that object as a token that can be fed into another compartment's module
+// map.
+// Only when the specified module has been loaded, analyzed, linked, and
+// instantiated (but no necessarily executed!) does it become possible to
+// distinguish properties that correspond to module exports.
+const createDeferredModule = () => {
+  let internalModule;
+  return freeze({
+    activate(module) {
+      internalModule = module;
+    },
+    module: new Proxy(immutableObject, {
+      get(ignore, name, target) {
```

Comment:
What about all the other handler traps, especially `getOwnPropertyDescriptor`? Does the `immutableObject` proxy target suppress all that? Once activated, how close can we come to the specified behavior of a module namespace object?

Reply:
I believe `immutableObject` does suppress all the other handlers. I’ll find the relevant bits in the spec and test.